Source/VMware.PSDesiredStateConfiguration/Functions/New-VmwDscConfiguration.ps1

```diff
@@ -14,90 +14,88 @@ Redistributions in binary form must reproduce the above copyright notice, this l
-                Property = @{
-                    Field = "Test field"
-                    Ensure = "Present"
-                }
-            )
-        }
-    )
-}
-
-.PARAMETER ConfigName
-Name of the dsc configuration to compile.
-
-.PARAMETER CustomParams
-The parameters for the dsc configuration.
-
-.PARAMETER ConfigurationData
-Configuration Data for the configuration.
-#>
 function New-VmwDscConfiguration {
+    <#
+    .SYNOPSIS
+        Compiles a DSC Configuration into a VmwDscConfiguration object, which contains
+        the name of the DSC Configuration and the DSC Resources defined in it.
+
+    .DESCRIPTION
+        Compiles a DSC Configuration into a VmwDscConfiguration object, which contains
+        the name of the DSC Configuration and the DSC Resources defined in it. The provided
+        PowerShell script file can contain multiple DSC Configurations in which case the cmdlet
+        returns an array of VmwDscConfiguration objects. If the ConfigurationName parameter is specified,
+        the cmdlet returns only one VmwDscConfiguration object - the one constructed from the specified
+        DSC Configuration.
+
+    .PARAMETER Path
+        Specifies a file path of a file that contains DSC Configurations. The file can contain multiple DSC Configurations
+        and for each one, a separate VmwDscConfiguration object is created. If ConfigurationData hashtable is defined in the provided
+        file, it is passed to each DSC Configuration defined in the file.
+
+    .PARAMETER ConfigurationName
+        Specifies the name of the DSC Configuration which should be compiled into a VmwDscConfiguration object. This parameter is applicable
+        only when multiple DSC Configurations are defined in the file and only one specific DSC Configuration should be compiled. If not specified,
+        all DSC Configurations in the file are compiled and returned as VmwDscConfiguration objects.
+
+    .PARAMETER CustomParams
+        Specifies the parameters of the DSC Configuration as a hashtable where each key is the parameter name and each value is the parameter value.
+        If the ConfigurationName parameter is specified, the hashtable is passed to the specific DSC Configuration.
+        If the ConfigurationName parameter is not specified, the hashtable is passed to all DSC Configurations defined in the PowerShell script file.
+
+    .EXAMPLE
+        Compiles the DSC Configurations defined in the vSphere_Config file located in the current directory.
+
+        PS> $dscConfigs = New-VmwDscConfiguration -Path .\vSphere_Config.ps1
+
+    .EXAMPLE
+        Compiles the DSC Configuration vSphereNode_Config defined in the vSphere_Config file located in the current directory
+        and passes the VMHostName parameter to the vSphereNode_Config DSC Configuration.
+
+        PS> $dscConfig = New-VmwDscConfiguration -Path .\vSphere_Config.ps1 -ConfigurationName 'vSphereNode_Config' -CustomParams @{ VMHostName = 'MyVMHost' }
```

Comment:
Currently, parameters are supported only for the configurations and not for a configuration file. 
In the case of a file with multiple configurations, CustomParams will be passed to each configuration. I think it would be better to support parameters on a configuration file rather than on a configuration.
Thus, -CustomParams would work for the passed configuration file